service/src/main/java/uk/nhs/adaptors/gp2gp/tasks/TaskExecutor.java

```diff
@@ -0,0 +1,8 @@
+package uk.nhs.adaptors.gp2gp.tasks;
+
+public interface TaskExecutor {
+
+    Class<? extends TaskDefinition> getTaskType();
+
```

Comment:
I would like to change this to return string, ticket does specify class returned here but I think it's not needed. also it makes it difficult to test and can't use mocks as factory need class type